.github/workflows/pull-request.yml

```diff
@@ -63,12 +63,14 @@ jobs:
       -
         uses: actions/checkout@v3
       -
-        uses: actions/setup-node@v3
+        uses: pnpm/action-setup@v4
+      -
+        uses: actions/setup-node@v4
         with:
-          node-version: "16.14.0"
-          cache: "npm"
+          node-version: "22.2.0"
+          cache: "pnpm"
       -
         name: Run prettier check
         run: |
-          npx prettier -v
-          npx prettier --check .
+          pnpx prettier -v
```

Comment:
hey @myieye , curious why some `pnpx` references were not changed to `pnpm exec` ?  The commit message I read didn't explain why.  Did you just miss these or were they left alone intentionally?

Reply:
I checked, and the difference between `pnpx` and `pnpm exec` is that `exec` runs a binary from your node_modules, e.g. `pnpm exec playwright` will run the Playwright binary that you have installed in your node_modules. Whereas `pnpx` is the equivalent of `pnpm dlx`, meaning download and execute. It will run a binary even if you *don't* have it in node_modules. If you wanted, for example, to run `prettier` *without* putting it in your devDependencies, then `pnpx` (or `pnpm dlx`) would work for that purpose.

Most of the time, though, you'd want `pnpm exec` since that way you know what version you're running. It's possible that `pnpx prettier` in CI could end up downloading a newer version of Prettier that includes breaking changes to defaults or config files, and then we'd get results on CI that are different than we got locally and be very confused. I think we want to use `pnpm exec` everywhere. I'll go ahead and change the remaining `pnpx` invocations to `pnpm exec`

(Incidentally, `pnpm exec` is optional as long as the name of what you're running doesn't conflict with a pnpm builtin command. So if you want to run a binary called `install` you'd have to run `pnpm exec install`, but you could safely replace `pnpm exec playwright` with `pnpm playwright` if you wanted to).

Reply:
Done in commit ed5d19434.